docs-src/docs/guides/walkthrough.md

```diff
@@ -244,11 +244,55 @@ Remember, we assume that you've cloned the [layer0-examples](https://github.com/
 We use these files to set up a Layer0 envrionment with Terraform:
 
 |Filename|Purpose|
-|----|----|
-|`terraform.tfvars`|Variables specific to the environment and guestbook application|
-|`Guestbook.Dockerrun.aws.json`|Template for running the guestbook application in a Layer0 environment|
-|`layer0.tf`|Provision Layer0 resources and populate variables in `Guestbook.Dockerrun.aws.json`|
+|---|---|
+|`Guestbook.Dockerrun.aws.json`|Template for running the Guestbook application|
+|`layer0.tf`|Provisions resources; populates resources in template files|
+|`terraform.tfstate`|Tracks status of deployment _(created and managed by Terraform)_|
+|`terraform.tfvars`|Variables specific to the environment and application(s)|
 
+### `Layer0.tf`: A Brief Aside
+
+Let's take a moment to look through the `layer0.tf` file. If you've followed along with the Layer0 CLI deployment above, it should be fairly easy to see how blocks in this file map to steps in the CLI process.
+
+When we began the CLI deployment, our first step was to create a load balancer:
+
+`l0 load balancer create demo-env`
+
+This command is recreated in `layer0.tf` like so:
+
+```
+resource "layer0_environment" "demo-env" {
+	name = "demo-env"
+}
+```
+
+The value of the `name` field inside the resource block is the name that maps to `demo-env` in the Layer0 CLI command. The `"demo-env"` in the resource declaration line is the identifier that Terraform will use to reference this resource.
+
+The next step in the CLI process would be to make a load balancer:
+
+`l0 loadbalancer create --port 80:80/http demo-env guestbook-lb`
+
+In `layer0.tf`:
+
+```
+resource "layer0_load_balancer" "guestbook-lb" {
+	name = "guestbook-lb"
+	environment = "${layer0_environment.demo-env.id}"
+	port {
+		host_port = 80
+		container_port = 80
+		protocol = "http"
+	}
+}
+```
+
+We use Terraform's interpolation syntax to discover and use the ID of our environment. The format is pretty simple:
+
+`${<resource_type>.<resource_identifier>.<property>}`
+
+And that's about all you need to be able to understand the file!
+
```

Comment:
maybe add a link here: "for more detailed information about layer0 resources in terraform, checkout http://layer0.ims.io/reference/terraform-plugin/"